File: src/main/java/io/github/dqualizer/dqlang/types/openapiadapter/field/FieldItem.java
package io.github.dqualizer.dqlang.types.openapiadapter.field;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Set;
import java.util.Map;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class FieldItem {

    private String path;
    @JsonProperty("operation_id")
    private String operationID;
    private String operation;
    private Set<Input> input;
    private Map<String, DataType> body;
    private Set<Map<String, Output>> output;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/archive/openapiadapter/dqlang/field/FieldItem.kt
package io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.field

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.ToString

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
class FieldItem {
    val path: String? = null

    @JsonProperty("operation_id")
    val operationID: String? = null
    val operation: String? = null
    val input: LinkedHashSet<io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.field.Input>? = null
    val body: Map<String, io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.field.DataType>? = null
    val output: LinkedHashSet<Map<String, io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.field.Output>>? = null
}
